contracts/prunto.sol

```diff
@@ -90,20 +102,40 @@ contract Prunto {
             request.amount,
             request.amount
         );
-        request.accepted = true;
+        request.accepted = true;   
+
+        emit acceptRequestEvent(request.requester, request.amount, "You successfully accepted a new request");
+
+        // delete request after accepting
+        deleteRequest(_id);
     }
 
+    // deny request at index @_id
     function denyRequest(uint256 _id) public {
         Request storage req = requests[msg.sender][_id];
         require(!req.accepted, "Request is already accepted.");
         require(!req.denied, "Request is already denied.");
         req.denied = true;
+
+        emit denyRequestEvent(req.requester, "You successfully denied a new request");
+
+        // delete Request after denying
+        deleteRequest(_id);
+    }
+
+    // delete request after accepting or rejecting
+    function deleteRequest(uint256 _id) public {
+        // replace the Request to delete with the last item in array
+        requests[msg.sender][_id] = requests[msg.sender][requests[msg.sender].length - 1];
+        requests[msg.sender].pop(); // delete the last item in the user's request list
```

Comment:
my intention was to keep the requests objects to display accepted and denied requests instead of deleting them from the blockchain.  I understand youre trying to fix the bug where the count of requests is off on the home page, but that is because i failed to filter the requests in the getRequestsLength method on line 9 of requests.js